projects/plugins/jetpack/modules/sharedaddy/sharing-sources.php

```diff
@@ -3041,3 +3081,125 @@ public function display_footer() {
 		endif;
 	}
 }
+
+/**
+ * Mastodon sharing service.
+ */
+class Share_Mastodon extends Sharing_Source {
+	/**
+	 * Service short name.
+	 *
+	 * @var string
+	 */
+	public $shortname = 'mastodon';
+
+	/**
+	 * Service icon font code.
+	 *
+	 * @var string
+	 */
+	public $icon = '\f10a';
+
+	/**
+	 * Constructor.
+	 *
+	 * @param int   $id       Sharing source ID.
+	 * @param array $settings Sharing settings.
+	 */
+	public function __construct( $id, array $settings ) { // phpcs:ignore Generic.CodeAnalysis.UselessOverridingMethod.Found
+		parent::__construct( $id, $settings );
+	}
+
+	/**
+	 * Service name.
+	 *
+	 * @return string
+	 */
+	public function get_name() {
+		return __( 'Mastodon', 'jetpack' );
+	}
+
+	/**
+	 * Get the markup of the sharing button.
+	 *
+	 * @param WP_Post $post Post object.
+	 *
+	 * @return string
+	 */
+	public function get_display( $post ) {
+		return $this->get_link(
+			$this->get_process_request_url( $post->ID ),
+			_x( 'Mastodon', 'share to', 'jetpack' ),
+			__( 'Click to share on Mastodon', 'jetpack' ),
+			'share=mastodon',
+			'sharing-mastodon-' . $post->ID
+		);
+	}
+
+	/**
+	 * Process sharing request. Add actions that need to happen when sharing here.
+	 *
+	 * @param WP_Post $post Post object.
+	 * @param array   $post_data Array of information about the post we're sharing.
+	 *
+	 * @return void
+	 */
+	public function process_request( $post, array $post_data ) {
+		if ( empty( $_POST['jetpack-mastodon-instance'] ) ) {
+			require_once WP_SHARING_PLUGIN_DIR . 'services/class-jetpack-mastodon-modal.php';
+			add_action( 'template_redirect', array( 'Jetpack_Mastodon_Modal', 'modal' ) );
+			return;
+		}
+
+		check_admin_referer( 'jetpack_share_mastodon_instance' );
+
+		$mastodon_instance = isset( $_POST['jetpack-mastodon-instance'] )
+			? trailingslashit( sanitize_text_field( wp_unslash( $_POST['jetpack-mastodon-instance'] ) ) )
+			: null;
+
+		$post_title = $this->get_share_title( $post->ID );
+		$post_link  = $this->get_share_url( $post->ID );
+		$post_tags  = $this->get_share_tags( $post->ID );
+
+		/**
+		 * Allow filtering the default message that gets posted to Mastodon.
+		 *
+		 * @module sharedaddy
+		 * @since $$next-version$$
+		 *
+		 * @param string  $share_url The default message that gets posted to Mastodon. This will get rawurlencoded.
```

Comment:
Why mention "This will get rawurlencoded"? Do filter hooks need to know how it's going to get processed to be sent to Mastodon?

Reply:
Removed in e2b292b2a0506946cd9fd78c1dc6421f26850886